File: assessments/javascript/models/restaurant.js

const Restaurant = class Restaurant {

  static all() {
    Store.restaurants
  }

  static findByName(name) {
    var restaurants = Store.restaurants
    let matches = (() => restaurants.filter(function (restaurant) {
      return restaurant.name == name
    }))()
    return matches
  }

  constructor(name) {
    this.name = name
    Store.restaurants.push(this)
  }
  }

File: assessments/object_relations/app/models/restaurant.rb
class Restaurant
  attr_accessor :customers, :reviews

  @@all = []

  def self.all
    @@all
  end

  def self.find_by_name(restaurant_name)
    #returns duplicates
    all.select {|restaurant| restaurant.name == restaurant_name}
  end

  def initialize(name)
    @@all << self
    @reviews = []
    @customers = []
  end



end
